src/oidc.rs

```diff
@@ -4,9 +4,26 @@ use std::convert::TryInto;
 use tame_oauth::Error;
 use url::form_urlencoded::Serializer;
 
+/// Request object sent in a token exchange request
+#[derive(serde::Deserialize, serde::Serialize, Debug)]
+pub struct TokenExchangeRequest {
+    /// Where to `POST` this request
+    pub uri: String,
+    /// Must be the same `redirect_uri` you used in the initial request
+    pub redirect_uri: String,
+    /// Identifies your application with the auth server
+    pub client_id: String,
+    /// The secret you don't share with anyone except the auth server
+    pub client_secret: Option<String>,
+    /// PKCE flow requires a code_verified to be present
+    pub code_verifier: Option<String>,
+    /// The auth_code you want to exchange for token(s)
+    pub auth_code: String,
+}
+
 /// This is the schema of the server's response.
 #[derive(serde::Deserialize, Debug)]
-struct TokenResponse {
+pub struct TokenExchangeResponse {
```

Comment:
How come this is public now?

Reply:
Also left-over from a previous installment. Removing. Nicely spotted!